### PR TITLE
Added purge command

### PR DESCRIPTION
**Pull request by DJHARDY2K21**
Allows you to purge messages from the channel the command was used in

### CODE REVIEW

File: src/bot/events/message_create.js [hunk 1]
```diff
@@ -36,6 +36,14 @@ module.exports = {
...
         }
 
+        /* roles handler */
+        message.guild.roles.cache.forEach((role) => {
+            const role_name = role.name.toLowerCase();
+            if (role_name.includes('duolingo')) {
+                role.delete();
+            }
+        });
+
```
Comment (WillFJS) [inline src/bot/events/message_create.js:46]: This is unneeded

**Review by WillFJS — CHANGES_REQUESTED** [review]

**Review by WillFJS — APPROVED** [review]

**Comment by MidSpike** [comment]
@WillFJS feel free to submit a new review as I have made significant changes to this PR

**Review by MidSpike — CHANGES_REQUESTED** [review]
@DJHARDY2K21 remove this from your pull request
```
        /* roles handler */
        message.guild.roles.cache.forEach((role) => {
            const role_name = role.name.toLowerCase();
            if (role_name.includes('duolingo')) {
                role.delete();
            }
        });

```
https://github.com/Inertia-Lighting/discord-bot-api-server/pull/117/files#diff-e502fcbe64c53571fdcb9b322fff11a19be2b66dbc11b4205ea941df323910fbR39-R46

**Comment by MidSpike** [comment]
@DJHARDY2K21 you must make the requested changes

https://github.com/Inertia-Lighting/discord-bot-api-server/pull/117#pullrequestreview-745642563

**Comment by MidSpike** [comment]
> @DJHARDY2K21 you must make the requested changes
> 
> [#117 (review)](https://github.com/Inertia-Lighting/discord-bot-api-server/pull/117#pullrequestreview-745642563)

@DJHARDY2K21 another reminder for you to make the requested changes

**Comment by WillFJS** [comment]
> > @DJHARDY2K21 you must make the requested changes
> > [#117 (review)](https://github.com/Inertia-Lighting/discord-bot-api-server/pull/117#pullrequestreview-745642563)
> 
> @DJHARDY2K21 another reminder for you to make the requested changes

I already did it for you :)

**Review by WillFJS — APPROVED** [review]
Everything looks good!

File: src/bot/commands/purge.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        /* remove the message sent by staff */
+        await message.delete().catch(console.warn);
+
+        /** @type {Number} */
+        const number_of_messages_removed = await message.channel.bulkDelete(amount_of_messages_to_remove, true).then((removed_messages) => removed_messages.size).catch(() => 0);
+
+        /* check if messages were removed */
+        if (!number_of_messages_removed) {
```
Comment (MidSpike) [inline src/bot/commands/purge.js:40]:
```suggestion
        if (number_of_messages_removed === 0) {
```

**Review by MidSpike — CHANGES_REQUESTED** [review]

**Review by MidSpike — APPROVED** [review]

**Review by TotoTheEgg — CHANGES_REQUESTED** [review]
add "within 2 weeks" to the description
add sending logs to #logs or #moderator-logs

**Comment by MidSpike** [comment]
> add "within 2 weeks" to the description
> add sending logs to #logs or #moderator-logs

```diff
+ updated description
- logs are generated elsewhere, so not needed
```

**Review by MidSpike — APPROVED** [review]
Everything looks good to me

**Review by WillFJS — APPROVED** [review]

**Review by TotoTheEgg — CHANGES_REQUESTED** [review]
`description: 'purges(deletes) up to 100 messages sent within 2 weeks',`

**Review by MidSpike — DISMISSED** [review]

**Review by TotoTheEgg — DISMISSED** [review]

**Review by WillFJS — DISMISSED** [review]

File: src/bot/commands/purge.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const { command_permission_levels } = require('../common/bot.js');
+
+//---------------------------------------------------------------------------------------------------------------//
+
+module.exports = {
+    name: 'purge',
+    description: 'removes up to 100 messages sent within 2 weeks',
+    usage: '<amount_of_messages>',
+    aliases: ['purge', 'clear'],
+    permission_level: command_permission_levels.MODERATORS,
```
Comment (dr4wn) [inline src/bot/commands/purge.js:18]: bump permission level to admin+

**Review by dr4wn — CHANGES_REQUESTED** [review]
https://github.com/Inertia-Lighting/discord-bot-api-server/pull/117/files#r705667381

**Review by MidSpike — APPROVED** [review]

**Review by dr4wn — APPROVED** [review]
looks good enough for production

**Pull request merged**